src/context/WalletProvider/WalletConnectV2/config.ts

```diff
@@ -1,11 +1,29 @@
-import { WalletConnectAdapter } from '@shapeshiftoss/hdwallet-walletconnect'
+import { WalletConnectV2Adapter } from '@shapeshiftoss/hdwallet-walletconnectv2'
+import type { EthereumProviderOptions } from '@walletconnect/ethereum-provider/dist/types/EthereumProvider'
+import { getConfig } from 'config'
 import { WalletConnectIcon } from 'components/Icons/WalletConnectIcon'
 import type { SupportedWalletInfo } from 'context/WalletProvider/config'
 
 export const WalletConnectV2Config: Omit<SupportedWalletInfo, 'routes'> = {
-  adapters: [WalletConnectAdapter],
+  adapters: [WalletConnectV2Adapter],
   supportsMobile: 'both',
   icon: WalletConnectIcon,
-  name: 'WalletConnect',
+  name: 'WalletConnectV2',
   description: 'v2',
 }
+
+export const walletConnectV2ProviderConfig: EthereumProviderOptions = {
+  projectId: getConfig().REACT_APP_WALLET_CONNECT_PROJECT_ID,
+  chains: [1],
+  optionalChains: [10, 56, 100, 137, 43114],
```

Comment:
Nitpick, can we either make this an IIFE and use asset data, or at least comment which chains these Ethereum network IDs refer to?

Reply:
Fixed this logic up in https://github.com/shapeshift/web/pull/5315/commits/bae3cf8f51be00329e9ead4e297b3eca523be7b3 to use the Viem chains.